### PR TITLE
plugins/lsp: add golangci-lint-langserver

### PR DESCRIPTION
**Pull request by alixinne**
Adds setup for [golangci-lint-langserver](https://github.com/nametake/golangci-lint-langserver).

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -234,6 +234,12 @@ with lib; let
       name = "gopls";
       description = "gopls for Go";
     }
+    {
+      name = "golangci-lint";
+      description = "golangci-lint-langserver for Go";
+      serverName = "golangci_lint_ls";
+      package = pkgs.golangci-lint-langserver;
```
Comment (traxys) [inline plugins/lsp/language-servers/default.nix:241]: I'd suggest going with the nvim-lspconfig name

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/lsp/language-servers/default.nix (hunk 1 above)
Comment (alixinne) [reply plugins/lsp/language-servers/default.nix:241]: Done in abb6e17f6832fa4ea53f7ec67c5eb25e392154d3

File: plugins/lsp/language-servers/default.nix [hunk 2]
```diff
@@ -234,6 +234,12 @@ with lib; let
       name = "gopls";
       description = "gopls for Go";
     }
+    {
+      name = "golangci-lint-ls";
+      description = "golangci-lint-langserver for Go";
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/default.nix:239]:
```suggestion
      description = "golangci-lint-ls for Go";
```

File: tests/test-sources/plugins/lsp/_lsp.nix [hunk 3]
```diff
@@ -107,6 +107,7 @@
           futhark-lsp.enable = true;
           gleam.enable = true;
           gopls.enable = true;
+          golangci-lint.enable = true;
```
Comment (GaetanLepage) [inline tests/test-sources/plugins/lsp/_lsp.nix:110]:
```suggestion
          golangci-lint-ls.enable = true;
```

File: plugins/lsp/language-servers/default.nix (hunk 2 above)
Comment (alixinne) [reply plugins/lsp/language-servers/default.nix]: Done in 2286bbb

File: tests/test-sources/plugins/lsp/_lsp.nix (hunk 3 above)
Comment (alixinne) [reply tests/test-sources/plugins/lsp/_lsp.nix]: Done in 2286bbb

**Review by GaetanLepage — APPROVED** [review]
LGTM thanks !

**Pull request merged**